runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -292,4 +295,92 @@ boolean shouldFilterByName(final String playerName)
 		}
 		return false;
 	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage chatMessage)
+	{
+		if (shouldCollapseMessageType(chatMessage.getType()))
+		{
+			duplicateChatCache.put(chatMessage.getName() + chatMessage.getMessage(), chatMessage.getMessageNode().getId());
+		}
+		if (duplicateChatCache.keySet().size() > config.maxTrackedChats())
+		{
+			cleanupChatTrackers();
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(final GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.HOPPING || event.getGameState() == GameState.LOGIN_SCREEN)
+		{
+			duplicateChatCache.clear();
+		}
+	}
+
+	private void cleanupChatTrackers()
+	{
+		log.debug("Cleaning up chat counts. Current size: {}", duplicateChatCache.keySet().size());
+		duplicateChatCache.keySet().removeIf(key -> duplicateChatCache.get(key).size() <= 1);
+		log.debug("After clearing messages with count == 1: {}", duplicateChatCache.keySet().size());
+		// Keep at least a buffer of 10 free spaces
+		if (duplicateChatCache.keySet().size() > config.maxTrackedChats() - 10)
+		{
+			duplicateChatCache.clear();
+		}
+	}
+
+	private boolean shouldFilterMessageType(ChatMessageType chatMessageType)
+	{
+		switch (chatMessageType)
+		{
+			case PUBLICCHAT:
+			case MODCHAT:
+			case AUTOTYPER:
+			case PRIVATECHAT:
+			case MODPRIVATECHAT:
+			case FRIENDSCHAT:
+				return true;
+			case LOGINLOGOUTNOTIFICATION:
+			default:
+				return false;
+		}
+	}
+
+	private boolean shouldCollapseMessageType(ChatMessageType chatMessageType)
+	{
+		switch (chatMessageType)
+		{
+			case ENGINE:
+			case GAMEMESSAGE:
+			case NPC_EXAMINE:
+			case OBJECT_EXAMINE:
+			case SPAM:
+				return true;
```

Comment:
What other chat types would you want to collapse? I believe the above are all of the "regular" game chats that are repetitive. Public chat is the only other kind of chat I would think would make sense to collapse.

Reply:
There is no way to turn off the collapsing for these types other than turning the plugin off.

Reply:
Ahh, my apologies. I just pushed a change that adds that as a configuration. I left that out because I initially made this as a separate plugin but I ended up merging it into the chatfilter plugin instead.